src/test/java/uk/gov/ch/oauth/identity/IdentityProviderTests.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.ch.oauth.identity;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+
+import static org.junit.Assert.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+public class IdentityProviderTests {
+
+
+    private IdentityProvider provider;
+
+    private EnvironmentReaderImpl reader;
+    private static final String code = "code";
+    private static final String state = "state";
+    private static final String scope = "scope";
+
+    @BeforeEach
+    public void init() {
+        reader = new EnvironmentReaderImpl();
+        provider = new IdentityProvider(reader);
+    }
+
+
+    @Test
+    @DisplayName("Test that getAuthorisationUrl(final String State) returns the expected result")
+    public void testThatGetAuthorisationUrlWithStateParameterReturnsTheExpectedUrl() {
+        String expectedAuthUrl = "/oauth2/authorise?client_id=test.apps.ch.gov.test&redirect_uri=/oauth2/user/callback&response_type=code&state=state";
```

Comment:
I don't like the use of these magic strings but can't think of a better alternative off of the top of my head.

Reply:
I will look into it again to see if there is a better alternative

Reply:
One solution I found was at least to make it more readable by using constants for each part of the url that is created using a getter in IdentityProvider so instead of:                          String expectedAuthUrl = "/oauth2/authorise?client_id=test.apps.ch.gov.test&redirect_uri=/oauth2/user/callback&response_type=code&state=state"        
 It would be: String expectedAuthUrl =  AUTHORISATIONURI + "?client_id=" + CLIENTID+ "&redirect_uri=" + REDIRECTURI + "&response_type=code&state=" + STATE;

Reply:
Isn't this getting quite close to simulating the behaviour that we're testing though?
The problem isn't that its not readable its that its fragile.
Again though. Minor problem and I don't really mind either way.